business/grafana_test.go

```diff
@@ -42,6 +42,7 @@ func TestGetGrafanaInfoDisabled(t *testing.T) {
 
 func TestGetGrafanaInfoExternal(t *testing.T) {
 	conf := config.NewConfig()
+	conf.ExternalServices.Grafana.InClusterURL = ""
```

Comment:
If users are not able to set an empty string in the CR for `in_cluser_url`, then they won't be able to reproduce this test scenario. And I am not 100% sure of the implications of it.

Reply:
But on the other side, the operator by default is always setting grafana's `in_cluster_url`. Therefore, it looks like there is a config that might not be achievable from the operator whilst it is from helm charts. Am I right?

In helm-charts but not with the operator:
```yaml
url: http://grafana.my-org.cat
in_cluster_url: ""
```

Reply:
Looking at this tells me it is OK that you can set inClusterURL to the same as URL. So rather than set it to empty string, just set it to the same as what you set it to URL and it should work.

https://github.com/kiali/kiali/blob/master/business/grafana.go#L127-L132

Reply:
BTW: it is not hard to change the operator to support setting this to empty string if that is what we need.

Reply:
> So rather than set it to empty string, just set it to the same as what you set it to URL and it should work.

However, we would have two tests recreating the very same scenario. That is why I prefer to keep the suite exactly the same but making explicit the empty string. Otherwise, we would be changing behavior.